crates/bitwarden-threading/src/cancellation_token.rs

```diff
@@ -0,0 +1,66 @@
+pub use tokio_util::sync::CancellationToken;
+
+#[cfg(target_arch = "wasm32")]
+pub mod wasm {
+    use wasm_bindgen::prelude::*;
+
+    use super::*;
+
+    #[wasm_bindgen]
+    extern "C" {
+        #[wasm_bindgen]
+        #[derive(Clone)]
```

Comment:
Question: Is this safe to be clonable when it contains a pointer to a JS value? I wonder how that works internally

Reply:
Yeah I was surprised to but it's perfectly safe. The imported type is just a wrapper around `JsValue` which itself is just a pointer to the actual value stored in a table

```
impl Clone for JsValue {
    #[inline]
    fn clone(&self) -> JsValue {
        unsafe {
            let idx = __wbindgen_object_clone_ref(self.idx);
            JsValue::_new(idx)
        }
    }
}
```

Reply:
Interesting, so Clone gives you two pointers to the same AbortController here right? I guess that makes sense following JS semantics of variable assignment.

```rust
let abort1 = AbortController::new();
let abort2 = abort1.clone();

abort1.abort(JsValue::from("Test reason"));
assert!(abort2.signal().aborted(), "Abort2 is also aborted")
```

Reply:
Yeah exactly, they act like an `Rc`. You can check the total amount of `JsValue` pointers in existence using `externref_heap_live_count`